nvflare/app_common/executors/multi_process_executor.py

```diff
@@ -145,14 +146,29 @@ def _initialize_multi_process(self, fl_ctx: FLContext):
 
         try:
             self.open_ports = get_open_ports(self.num_of_processes * 3)
-
+            client_name = fl_ctx.get_identity_name()
+            job_id = fl_ctx.get_job_id()
+
+            engine = fl_ctx.get_engine()
+            if isinstance(engine, SimulatorClientRunManager):
+                simulator_engine = True
+            else:
+                simulator_engine = False
             command = (
                 self.get_multi_process_command()
                 + " -m nvflare.private.fed.app.client.sub_worker_process"
```

Comment:
But then this line is arguable.
If this "nvflare/app_common/executors/multi_process_executor.py" rely on "nvflare.private.fed.app.client.sub_worker_process", we should put this class in private?

Reply:
changed the way to determine simulator mode. removed this import.